### PR TITLE
Fix analyzer warning in test

### PR DESCRIPTION
**Pull request by krisgiesing**

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
wait wait

**Comment by Hixie** [comment]
what's the analyzer warning?

**Comment by Hixie** [comment]
There shouldn't be any warning here. Does the analyzer in Atom not support generic methods?

**Comment by devoncarew** [comment]
I see the warning for 1.17.0 SDKs:

screen shot 2016-05-03 at 1 23 03 pm

**Comment by Hixie** [comment]
So the problem is that Atom's analyzer configuration doesn't have generic method support enabled, because we don't have a way to enable that across all our packages yet without lots of separate redundant files in the repo.

**Comment by Hixie** [comment]
If I understand correctly, according to @pq we can put an .analysis_options file in the root of the repo and it'll cover every package in our repo. That would be a better fix.

**Comment by pq** [comment]
Yep.  That's right.  Having a single `.analysis_options` in (say) `flutter/packages` might get us a lot of the way to a single source of truth.  (We can consume this in `flutter analyze` too obviously.)

**Comment by krisgiesing** [comment]
So I should abandon this PR?

**Comment by Hixie** [comment]
I'd replace it with an `.analysis_options` file in the root of the flutter repo that is a copy of the `flutter_analysis_options` file but with the bits that make Atom angry commented out.

**Comment by pq** [comment]
The down side of the root analysis options approach is that it's contents will need to be duplicated in all the places it's overridden.  For example, I see the following in the stocks example `.analysis_options`:

```
analyzer:
  exclude:
    - 'lib/i18n/stock_messages_*.dart'
```

The presence of this file would override the root one and the stocks example would get no config besides the exclude specified here.

**Comment by krisgiesing** [comment]
I copied the aforementioned file to ```packages``` and indeed the warning went away. I did have to remove the lint warning about documenting public API methods; I think even ```flutter analyze``` handles that case differently.

The stocks example probably doesn't need its own ```.analysis_options``` file. I tried removing it and didn't get any extra warnings. There are a few others, though, that I haven't looked at.

**Comment by krisgiesing** [comment]
I don't suppose there is a way to include one options file in another?

**Comment by krisgiesing** [comment]
I ended up leaving the other options files in place; it seems safe to leave them for a later cleanup pass.

**Comment by abarth** [comment]
LGTM

File: packages/.analysis_options [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Specify analysis options.
```
Comment (Hixie) [inline packages/.analysis_options]: Why is this in `packages/` rather than in the root? We have dart in other subdirectories of this repo.

Comment (krisgiesing) [inline packages/.analysis_options]: I misunderstood the comments in this thread. We can move it.

Comment (krisgiesing) [inline packages/.analysis_options]: New pull request #3750

**Pull request merged**